src/components/hosts/HostStatus.tsx

```diff
@@ -184,34 +252,47 @@ const HostStatus: React.FC<HostStatusProps> = ({
     (status === 'added-to-existing-cluster' && 'Finish in console') ||
     undefined;
 
+  const titleWithProgress = withProgress(title, stageNumber, stages.length, status);
+
   return (
-    <>
-      <Popover
-        headerContent={<div>{title}</div>}
-        bodyContent={
-          <HostStatusPopoverContent
-            host={host}
-            validationsInfo={validationsInfo}
-            onEditHostname={toggleHostname}
-          />
-        }
-        footerContent={<HostStatusPopoverFooter host={host} />}
-        minWidth="30rem"
-        maxWidth="50rem"
-        hideOnOutsideClick={!keepOnOutsideClick}
-        zIndex={300}
-      >
-        <Button variant={ButtonVariant.link} isInline>
-          {icon} {title}{' '}
-          {['installing', 'installing-in-progress', 'error', 'cancelled'].includes(status) && (
-            <>
-              {getHostProgressStageNumber(host)}/{hostProgressStages.length}
-            </>
-          )}
-        </Button>
-      </Popover>
-      {sublabel && <div className="hosts-table-sublabel">{sublabel}</div>}
-    </>
+    <Flex alignItems={{ default: 'alignItemsCenter' }}>
+      <FlexItem className={'pf-u-mr-xs'}>{icon}</FlexItem>
+
+      <Flex direction={{ default: 'column' }}>
+        <RenderIfElse
+          condition={!sublabel}
+          truthy={
+            <WithHostStatusPopover
+              hideOnOutsideClick={!keepOnOutsideClick}
+              host={host}
+              onEditHostName={toggleHostname}
+              title={title}
+              validationsInfo={validationsInfo}
+            >
+              {titleWithProgress}
+            </WithHostStatusPopover>
+          }
+          falsy={<FlexItem className={'pf-u-mb-0'}>{titleWithProgress}</FlexItem>}
+        />
+
+        <RenderIf condition={Boolean(sublabel)}>
+          <FlexItem
+            className={'pf-u-font-size-xs'}
+            style={{ marginTop: 'calc(-1 * var(--pf-l-flex--spacer--xs))' }}
```

Comment:
Dear reviewers,
This is the only place I had to tweak the style, honestly, I see no benefit in creating a CSS class only for this case.
IMHO, should this "tweak" become something widely used in our codebase, only then would be worth creating a CSS class for it.

Reply:
It's a matter of code style. So far we have created a class for everything, avoiding direct styling. My preference is to keep that approach.

Reply:
What's the reason to use classes for styling everything? The whole idea behind the CSS classes is to allow to reuse of style rules. In this case, there's nothing to be reused.

Reply:
Honestly I'd get rid of the negative margin altogether and keep the styling intact. Alternatively we'd have to tweak the line heights for sublabel and label but IMHO it is not needed.

Reply:
So what do we want? 
Keep using CSS classes even for non-reusable rules? How many times are you gonna write a CSS class that does `margin` this `padding` that just to override it in a specific place?

Right now we have something that is very close to the mock, is it worth throwing that away just because there's an explicit coding convention around the CSS?

Reply:
Currently looks like this:
![image](https://user-images.githubusercontent.com/77008341/123082427-aa3efd80-d427-11eb-89f6-efb25ac4df5b.png)

Without the margin-top override looks like this:
![image](https://user-images.githubusercontent.com/77008341/123082268-8380c700-d427-11eb-8bf6-7471a5586996.png)

Reply:
In general I am not opposed to adding style to a component directly.
I like to follow these rules:
1/ use vanilla PF as much as possible
2/ define classes for reused overrides to avoid duplication
3/ small tweaks using 'style' in single place are fine as it simplifies eventual removing, we don't have to open another file to see the tweaks applied, no need to invent className name

Reply:
@jtomasek then I think we are squared here, I followed the same principle.